dev-libs/libuv/patches/libuv-1.21.0.patchset

```diff
@@ -0,0 +1,496 @@
+ AM_CONDITIONAL([MSYS],     [AS_CASE([$host_os],[msys*],         [true], [false])])
+ AM_CONDITIONAL([NETBSD],   [AS_CASE([$host_os],[netbsd*],       [true], [false])])
+ AM_CONDITIONAL([OPENBSD],  [AS_CASE([$host_os],[openbsd*],      [true], [false])])
+@@ -66,6 +72,9 @@ AM_CONDITIONAL([WINNT],    [AS_CASE([$host_os],[mingw*],        [true], [false])
+ AS_CASE([$host_os],[mingw*], [
+     LIBS="$LIBS -lws2_32 -lpsapi -liphlpapi -lshell32 -luserenv -luser32"
+ ])
++AS_CASE([$host_os],[haiku*], [
++    LIBS="$LIBS -lroot -lbe -lnetwork -lshared"
++])
+ AS_CASE([$host_os], [netbsd*], [AC_CHECK_LIB([kvm], [kvm_open])])
+ AC_CHECK_HEADERS([sys/ahafs_evProds.h])
+ AC_CONFIG_FILES([Makefile libuv.pc])
+diff --git a/include/uv/errno.h b/include/uv/errno.h
+index 8eeb95d..00d6d5c 100644
+--- a/include/uv/errno.h
++++ b/include/uv/errno.h
+@@ -248,7 +248,7 @@
+ # define UV__ENOENT (-4058)
+ #endif
+ 
+-#if defined(ENOMEM) && !defined(_WIN32)
++#if defined(ENOMEM) && !defined(_WIN32) && !defined(__HAIKU__)
+ # define UV__ENOMEM UV__ERR(ENOMEM)
+ #else
+ # define UV__ENOMEM (-4057)
+diff --git a/include/uv/unix.h b/include/uv/unix.h
+index 74a0d64..3fc8229 100644
+--- a/include/uv/unix.h
++++ b/include/uv/unix.h
+@@ -62,7 +62,7 @@
+       defined(__OpenBSD__)         || \
+       defined(__NetBSD__)
+ # include "uv/bsd.h"
+-#elif defined(__CYGWIN__) || defined(__MSYS__)
++#elif defined(__CYGWIN__) || defined(__MSYS__) || defined(__HAIKU__)
+ # include "uv/posix.h"
+ #endif
+ 
+diff --git a/src/unix/bsd-ifaddrs.c b/src/unix/bsd-ifaddrs.c
+index 0d02154..d0b3ba1 100644
+--- a/src/unix/bsd-ifaddrs.c
++++ b/src/unix/bsd-ifaddrs.c
+@@ -25,12 +25,20 @@
+ #include <errno.h>
+ #include <stddef.h>
+ 
++#if !defined(__HAIKU__)
+ #include <ifaddrs.h>
++#else
++#include <bsd/ifaddrs.h>
++#endif
+ #include <net/if.h>
+ #if !defined(__CYGWIN__) && !defined(__MSYS__)
+ #include <net/if_dl.h>
+ #endif
+ 
++#ifndef IFF_RUNNING
++#define IFF_RUNNING 0x0001 
++#endif
++
+ static int uv__ifaddr_exclude(struct ifaddrs *ent, int exclude_type) {
+   if (!((ent->ifa_flags & IFF_UP) && (ent->ifa_flags & IFF_RUNNING)))
+     return 1;
+@@ -45,7 +53,7 @@ static int uv__ifaddr_exclude(struct ifaddrs *ent, int exclude_type) {
+   if (exclude_type == UV__EXCLUDE_IFPHYS)
+     return (ent->ifa_addr->sa_family != AF_LINK);
+ #endif
+-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__)
++#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__HAIKU__)
+   /*
+    * On BSD getifaddrs returns information related to the raw underlying
+    * devices.  We're not interested in this information.
+diff --git a/src/unix/core.c b/src/unix/core.c
+index c2e7bd7..74061c0 100644
+--- a/src/unix/core.c
++++ b/src/unix/core.c
+@@ -56,6 +56,19 @@
+ # endif
+ #endif
+ 
++#ifdef __HAIKU__
++# include <fcntl.h>
++# if defined(O_CLOEXEC)
++#  define UV__O_CLOEXEC O_CLOEXEC
++# endif
++# if !defined(FIONCLEX)
++#  define FIONCLEX        0x5450
```

Comment:
Please avoid using magic numbers that doesn't have a definite meaning in Haiku. Especially when you are passing them to `ioctl()`.